src/status_im/ui/screens/wallet/send/views/recipient.cljs

```diff
@@ -0,0 +1,231 @@
+(ns status-im.ui.screens.wallet.send.views.recipient
+  (:require [status-im.utils.ethereum.core :as ethereum]
+            [status-im.utils.ethereum.tokens :as tokens]
+            [reagent.core :as reagent]
+            [status-im.ui.screens.wallet.components.views :as wallet.components]
+            [status-im.i18n :as i18n]
+            [status-im.ui.screens.wallet.send.views.common :as common]
+            [re-frame.core :as re-frame]
+            [status-im.ui.components.colors :as colors]
+            [status-im.ui.components.list.styles :as list.styles]
+            [status-im.ui.components.react :as react]
+            [status-im.ui.screens.chat.photos :as photos]
+            [status-im.ui.components.icons.vector-icons :as vector-icons]
+            [status-im.ui.screens.wallet.send.events :as events]
+            [status-im.utils.utils :as utils]
+            [clojure.string :as string]
+            [status-im.ui.screens.wallet.send.styles :as styles]
+            [status-im.ui.components.tooltip.views :as tooltip]
+            [status-im.utils.ethereum.ens :as ens]
+            [status-im.ui.components.list.views :as list]))
+
+(defn simple-tab-navigator
+  "A simple tab navigator that that takes a map of tabs and the key of
+  the starting tab
+
+  Example:
+  (simple-tab-navigator
+   {:main {:name \"Main\" :component (fn [] [react/text \"Hello\"])}
+    :other {:name \"Other\" :component (fn [] [react/text \"Goodbye\"])}}
+   :main)"
+  [tab-map default-key]
+  {:pre [(keyword? default-key)]}
+  (let [tab-key (reagent/atom default-key)]
+    (fn [tab-map _]
+      (let [tab-name @tab-key]
+        [react/view {:flex 1}
+         ;; tabs row
+         [react/view {:flex-direction :row}
+          (map (fn [[key {:keys [name component]}]]
+                 (let [current? (= key tab-name)]
+                   ^{:key (str key)}
+                   [react/view {:flex             1
+                                :background-color colors/black-transparent}
+                    [react/touchable-highlight {:on-press #(reset! tab-key key)
+                                                :disabled current?}
+                     [react/view {:height              44
+                                  :align-items         :center
+                                  :justify-content     :center
+                                  :border-bottom-width 2
+                                  :border-bottom-color (if current? colors/white colors/transparent)}
+                      [react/text {:style {:color     (if current? colors/white colors/white-transparent)
+                                           :font-size 15}} name]]]]))
+               tab-map)]
+         (when-let [component-thunk (some-> tab-map tab-name :component)]
+           [component-thunk])]))))
+
+(defn open-qr-scanner [chain text-input transaction]
+  (.blur @text-input)
+  (re-frame/dispatch [:navigate-to :recipient-qr-code
+                      {:on-recipient
```

Comment:
so here it should dispatch event and pass db into extract-qr-code-details

Reply:
yeah, that should work fine